src/graphics/pyro/pyro.h

```diff
@@ -185,5 +161,185 @@ class CPyro
+};
+
+class CWinPyro : public CPyro
+{
+public:
+    CWinPyro(CObject *obj);
+    Error IsEnded() override;
+    void UpdateEffect() override;
+};
+
+class CSpiderPyro : public CPyro
+{
+public:
+    CSpiderPyro(CObject *obj);
+    bool EventProcess(const Event& event) override;
+    void AfterCreate() override;
+};
+
+class CEggPyro : public CPyro
+{
+public:
+    CEggPyro(CObject *obj);
+    bool EventProcess(const Event& event) override;
+    void AfterCreate() override;
+};
+
+class CDeadGPyro : public CPyro
+{
+public:
+    CDeadGPyro(CObject *obj);
+    void AfterCreate() override;
+};
+
+class CDeadWPyro : public CPyro
+{
+public:
+    CDeadWPyro(CObject *obj);
+    void AfterCreate() override;
+};
+
+class CFindingPyro : public CPyro
+{
+public:
+    CFindingPyro(CObject *obj);
+    void AfterCreate() override;
+    void UpdateEffect() override;
+};
+
+class CSquashPyro : public CPyro
+{
+public:
+    CSquashPyro(CObject *obj);
+    void AfterCreate() override;
+    void UpdateEffect() override;
+    void AfterEnd() override;
+};
+
+class CFragVPyro : public CPyro
+{
+public:
+    CFragVPyro(CObject *obj);
+    bool EventProcess(const Event& event) override;
+    void AfterCreate() override;
+};
+
+class CBurnPyro : public CPyro
+{
+    bool m_organicBurn;
+public:
+    CBurnPyro(CObject *obj);
+    void AfterCreate() override;
+    void UpdateEffect() override;
+
+    //! Starts a vehicle fire
+    void        BurnStart();
+    //! Adds a part move
+    void        BurnAddPart(int part, Math::Vector pos, Math::Vector angle);
+    //! Advances of a vehicle fire
+    void        BurnProgress();
+    //! Indicates whether a part should be retained
+    bool        BurnIsKeepPart(int part);
+    //! Ends the fire of an insect or a vehicle
+    void        AfterEnd() override;
+};
+
+
+/**
+ * \enum FragExploOrShotPyroType
+ * \brief Type of CFragExploOrShotPyro
+ */
+enum FragExploOrShotPyroType
+{
+    PT_FRAGT,               //! < fragmentation of technical object
+    PT_FRAGO,               //! < fragmentation of organic object
+    PT_FRAGW,               //! < fragmentation of object under water
+    PT_EXPLOT,              //! < explosion of technical object
+    PT_EXPLOO,              //! < explosion of organic object
+    PT_EXPLOW,              //! < explosion of object under water
+    PT_SHOTT,               //! < hit technical object
+    PT_SHOTH,               //! < hit human
+    PT_SHOTM,               //! < hit queen
+    PT_SHOTW,               //! < hit under water (TODO: check if unused)
+};
+
+class CFragExploOrShotPyro : public CPyro
+{
+    float m_force;
+    FragExploOrShotPyroType m_type;
+public:
+    CFragExploOrShotPyro(FragExploOrShotPyroType type, CObject *obj, float force = 1.0f);
+    void AfterCreate() override;
+    bool EventProcess(const Event& event) override;
+    void UpdateEffect() override;
+
+    //! Starts the explosion of a vehicle
+    void        ExploStart();
```

Comment:
ExploStart() probably should not be public. Or is it used outside CFragExploOrShotPyro?